index.js

```diff
@@ -179,10 +179,10 @@ const handlePromise = async ({spawned, options, stdioStreamsGroups, originalStre
 	const stdio = stdioResults.map(stdioResult => handleOutput(options, stdioResult));
 	const all = handleOutput(options, allResult);
 
-	if (error || exitCode !== 0 || signal !== null) {
+	if ('error' in errorInfo) {
```

Comment:
Allows handling `childProcess.emit('error')`, which produces an error which is `undefined`.